lib/qa/authorities/authority_with_sub_authority.rb

```diff
@@ -1,12 +1,18 @@
 module Qa::Authorities
-  class AuthorityWithSubAuthority < Base
-    attr_reader :sub_authority
+  module AuthorityWithSubAuthority
 
     # Registers the authority and its sub-authority if it has one
-    def initialize(sub_authority=nil)
-      @sub_authority = sub_authority
-      raise Qa::MissingSubAuthority, "No sub-authority provided" if sub_authority.nil?
-      raise Qa::InvalidSubAuthority, "Unable to initialize sub-authority #{sub_authority} for #{self.class.name}" unless sub_authorities.include?(sub_authority)
+    def new(sub_authority=nil)
+      raise "Initializing with as sub authority is removed. use #{self.class}.factory(#{sub_authority.inspect}) instead"
+    end
+
+    def factory(sub_authority)
```

Comment:
The name `factory` is not very clear, especially since this is a module method, and so it will be mixed into other classes.  Maybe `new_sub_authority`?